modules/k8s-cluster/main.tf

```diff
@@ -51,37 +51,58 @@ resource "aws_cloudwatch_log_group" "eks" {
   }
 }
 
-# As per https://docs.aws.amazon.com/eks/latest/userguide/launch-workers.html
-resource "aws_cloudformation_stack" "worker-nodes" {
-  name          = "${var.cluster_name}-worker-nodes"
-  template_body = "${file("${path.module}/data/nodegroup.yaml")}"
-  capabilities  = ["CAPABILITY_IAM"]
-
-  parameters = {
-    NodeImageId                         = "/aws/service/eks/optimized-ami/${var.worker_eks_version}/amazon-linux-2/recommended/image_id"
-    ClusterName                         = "${var.cluster_name}"
-    ClusterControlPlaneSecurityGroup    = "${aws_security_group.controller.id}"
-    NodeGroupName                       = "worker"
-    NodeAutoScalingGroupMinSize         = "0"
-    NodeAutoScalingGroupDesiredCapacity = "0"
-    NodeAutoScalingGroupMaxSize         = "0"
-    NodeInstanceType                    = "${var.worker_instance_type}"
-    NodeVolumeSize                      = "40"
-    BootstrapArguments                  = "--kubelet-extra-args \"--node-labels=node-role.kubernetes.io/worker --event-qps=0\""
-    VpcId                               = "${var.vpc_id}"
-    Subnets                             = "${join(",", var.private_subnet_ids)}"
+data "aws_iam_policy_document" "worker-nodes-assume-role-policy" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals = {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
   }
+}
 
-  timeouts {
-    create = "30m"
+resource "aws_iam_role" "worker-nodes-role" {
```

Comment:
it feels like this should live in `iam.tf` instead of here in between the nodegroup instantiations?